inference-engine/src/transformations/src/transformations/common_optimizations/nearest_neighbor_umpsampling_fusion.cpp

```diff
@@ -0,0 +1,322 @@
+    //      6) all elements of 'mul_const' are equal to 1.0.
+    //
+    // Such subgraph can be replaced by the Interpolate node with
+    //      1) mode='nearest' and shape_calculation_mode='scales';
+    //      2) 'sizes' input as a constant with the value [newD_1, newD_2, ..., newD_i, ..., newD_{r - 2}];
+    //      3) 'scales' input as a constant with the value [S_1, S_2, ..., S_i, ..., S_{r - 2}];
+    //      4) 'axes' input as a constant with the value [1, 2, ..., r - 2].
+    //
+    // Of course, the replacement shouldn't be done, if all S_i are equal to 1.
+    auto input = ngraph::pattern::any_input();
+    auto concat_1 = pattern::wrap_type<opset8::Concat>();
+    auto concat_2 = pattern::wrap_type<opset8::Concat>();
+    auto reshape_1 = pattern::wrap_type<opset8::Reshape>({input, concat_1});
+    auto mul_const = pattern::wrap_type<opset8::Constant>();
+    auto mul = pattern::wrap_type<opset8::Multiply>({reshape_1, mul_const});
+    auto reshape_2 = pattern::wrap_type<opset8::Reshape>({mul, concat_2});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto &pattern_to_output = m.get_pattern_value_map();
+
+        const auto reshape_2_node = std::dynamic_pointer_cast<opset8::Reshape>(pattern_to_output.at(reshape_2).get_node_shared_ptr());
+        const auto mul_node = std::dynamic_pointer_cast<opset8::Multiply>(pattern_to_output.at(mul).get_node_shared_ptr());
+
+        if (!reshape_2_node || !mul_node || mul_node->get_input_partial_shape(1).is_dynamic()) return false;
+
+        const auto mul_const_node = std::dynamic_pointer_cast<opset8::Constant>(mul_node->input_value(1).get_node_shared_ptr());
+        if (!mul_const_node) return false;
+
+        const auto reshape_1_node = std::dynamic_pointer_cast<opset8::Reshape>(pattern_to_output.at(reshape_1).get_node_shared_ptr());
+        if (!reshape_1_node || reshape_1_node->get_input_partial_shape(0).is_dynamic()) return false;
+
+        uint64_t input_rank = static_cast<uint64_t>(reshape_1_node->get_input_partial_shape(0).rank().get_length());
+        const auto mul_const_shape = mul_const_node->get_output_shape(0);
+        const auto scales = get_scales_from_mul_const_shape(mul_const_shape, input_rank);
+        if (scales.empty() || std::all_of(scales.begin(), scales.end(), [](float s) { return s == 1.0f;})) { return false; }
+
+        const auto mul_const_value = mul_const_node->cast_vector<float>();
+        if (std::any_of(mul_const_value.begin(), mul_const_value.end(), [](float x){ return x != 1.0f; })) { return false; }
+
+        const auto concat_1_node = std::dynamic_pointer_cast<opset8::Concat>(reshape_1_node->input_value(1).get_node_shared_ptr());
+        if (!concat_1_node) return false;
+
+        const auto input_shape = reshape_1_node->get_input_shape(0);
+        if (!check_concat_1(concat_1_node, input_shape)) return false;
```

Comment:
Looks like this transformation is not dynamism friendly. So we can put it inside MOC pipeline but under `m_use_shapes == true` statement, so it won't be executed in MO by default but inside plugins it will be executed if shapes are static.

@jane-intel FYI ^^^

Reply:
Done.